### PR TITLE
Active network management (draft)

### PR DESCRIPTION
**Pull request by teytaud**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context / Related issue

Active network management is one of the key challenges for the 21st century.

## How Has This Been Tested (if it applies)

Not tested at all, this might be bs. Will test properly later.

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the [**CONTRIBUTING**](https://facebookresearch.github.io/nevergrad/contributing.html) document and completed the CLA (see [**CLA**](https://facebookresearch.github.io/nevergrad/contributing.html#contributor-license-agreement-cla)).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

**Comment by teytaud** [comment]
Hum actually it should be possible to do this in a generic  manner.

File: nevergrad/functions/gymanm/gym_anm.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# LICENSE file in the root directory of this source tree.
+
+
+import numpy as np
+from nevergrad.parametrization import parameter
+from ..base import ExperimentFunction
+import gym
+
+# pylint: disable=too-many-locals,too-many-statements
+
```
Comment (jrapin) [inline nevergrad/functions/gymanm/gym_anm.py:13]:
```suggestion
```

File: nevergrad/benchmark/experiments.py [hunk 2]
```diff
@@ -1073,6 +1074,24 @@ def rocket(seed: tp.Optional[int] = None) -> tp.Iterator[Experiment]:
...
+    Sequential or 30 workers."""
+    func = GymAnm()
+    seedg = create_seed_generator(seed)
+    optims = get_optimizers("basics", seed=next(seedg))
+    for budget in [25, 50, 100, 200, 400, 800, 1600]:
+        for num_workers in [1, 30]:
+            if num_workers < budget:
+                for algo in optims:
+                    xp = Experiment(func, algo, budget, num_workers=num_workers, seed=next(seedg))
+                    # skip_ci(reason="Too slow")
```
Comment (jrapin) [inline nevergrad/benchmark/experiments.py:1090]:
```suggestion
```

File: nevergrad/functions/gymanm/gym_anm.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+def gym_anm(x: np.ndarray):
+
+    env = gym.make("gym_anm:ANM6Easy-v0")
+    env.seed(0)
+    _ = env.reset()  # output value = "o"
+
+    reward = 0.0
+    for i in range(100):
```
Comment (jrapin) [inline nevergrad/functions/gymanm/gym_anm.py:27]:
```suggestion
    for i, val in enumerate(x):
```

File: nevergrad/functions/gymanm/gym_anm.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        a = env.action_space.sample()
```
Comment (jrapin) [inline nevergrad/functions/gymanm/gym_anm.py:28]: I suppose this needs to be removed?
```suggestion
```

File: nevergrad/functions/gymanm/test_gym_anm.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import numpy as np
+from nevergrad.common import testing
+from . import gym_anm
+
+
+def test_gym_anm() -> None:
+    with testing.skip_error_on_systems(OSError, systems=("Windows",)):
```
Comment (jrapin) [inline nevergrad/functions/gymanm/test_gym_anm.py:12]: what was the problem on windows?

File: nevergrad/functions/gymanm/test_gym_anm.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+        func = gym_anm.GymAnm()
+        x = 0 * np.random.rand(func.dimension)
```
Comment (jrapin) [inline nevergrad/functions/gymanm/test_gym_anm.py:14]:
```suggestion
        x = np.random.zeros(func.dimension)
```

File: nevergrad/functions/gymanm/test_gym_anm.py [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+        value = func(x)  # should not touch boundaries, so value should be < np.inf
```
Comment (jrapin) [inline nevergrad/functions/gymanm/test_gym_anm.py:15]:
```suggestion
        value = func(x)
```

File: nevergrad/functions/gymanm/test_gym_anm.py [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+        np.testing.assert_almost_equal(value, 1e20)
```
Comment (jrapin) [inline nevergrad/functions/gymanm/test_gym_anm.py:16]: looks like this should be equal in this case
```suggestion
        assert value == 1e20
```

File: requirements/bench.txt [hunk 9]
```diff
@@ -5,6 +5,7 @@ lpips==0.1.3
 opencv-python==4.1.2.30
 matplotlib>=2.2.3
 gym>=0.12.1
+gym-anm
```
Comment (jrapin) [inline requirements/bench.txt:8]: please add a version

File: nevergrad/functions/gymanm/test_gym_anm.py (hunk 5 above)
Comment (teytaud) [reply nevergrad/functions/gymanm/test_gym_anm.py:12]: It tells me than gym-anm is not present. No idea how this can happen on Windows and not Linux.

Comment (teytaud) [reply nevergrad/functions/gymanm/test_gym_anm.py:12]: Actually this fails at the import. Need to disable entirely this on Windows seemingly.

**Comment by teytaud** [comment]
Closed, branch will be deleted unless there is a strong argument against pr/1081.

**Pull request closed without merging**